platform/feature-api/communication/remote-pod/src/remote.ts

```diff
@@ -39,64 +39,71 @@ import { RequestListener } from "http";
+    select(matcher: Partial<Matcher>): ValueBackendEndpointSpec<Quad[]>;
+    match(matcher: Partial<Matcher>): ValueBackendEndpointSpec<Quad[]>;
+    add(...quads: Quad[]): ValueBackendEndpointSpec<void>;
+    delete(...quads: Quad[]): ValueBackendEndpointSpec<void>;
+    has(...quads: Quad[]): ValueBackendEndpointSpec<boolean>;
 }>;
 
-type PolyOutEndpoint = ObjectEndpointSpec<{
-    readDir(path: string): ValueEndpointSpec<Entry[]>;
-    readFile(path: string, options?: EncodingOptions): ValueEndpointSpec<string | Uint8Array>;
-    writeFile(path: string, content: string, options: EncodingOptions): ValueEndpointSpec<void>;
-    stat(path: string): ValueEndpointSpec<Stats>;
-    fetch(input: string, init: RequestInit): ValueEndpointSpec<Response>;
-    importArchive(url: string): ValueEndpointSpec<string>;
-    removeArchive(fileId: string): ValueEndpointSpec<void>;
+type PolyOutBackend = ObjectBackendEndpointSpec<{
+    readDir(path: string): ValueBackendEndpointSpec<Entry[]>;
+    readFile(
+        path: string,
+        options?: EncodingOptions
+    ): ValueBackendEndpointSpec<string | Uint8Array>;
+    writeFile(
+        path: string,
+        content: string,
+        options: EncodingOptions
+    ): ValueBackendEndpointSpec<void>;
+    stat(path: string): ValueBackendEndpointSpec<Stats>;
+    fetch(input: string, init: RequestInit): ValueBackendEndpointSpec<Response>;
+    importArchive(url: string): ValueBackendEndpointSpec<string>;
+    removeArchive(fileId: string): ValueBackendEndpointSpec<void>;
 }>;
 
-type PolyLifecycleEndpoint = ObjectEndpointSpec<{
-    listFeatures(): ValueEndpointSpec<Record<string, boolean>>;
-    startFeature(id: string, background: boolean): ValueEndpointSpec<void>;
+type PolyLifecycleBackend = ObjectBackendEndpointSpec<{
+    listFeatures(): ValueBackendEndpointSpec<Record<string, boolean>>;
+    startFeature(id: string, background: boolean): ValueBackendEndpointSpec<void>;
 }>;
 
-type PolyNavEndpoint = ObjectEndpointSpec<{
-    openUrl(url: string): ValueEndpointSpec<void>;
-    setActiveActions(actions: string[]): ValueEndpointSpec<void>;
-    setTitle(title: string): ValueEndpointSpec<void>;
-    pickFile(type?: string): ValueEndpointSpec<ExternalFile | null>;
+type PolyNavBackend = ObjectBackendEndpointSpec<{
+    openUrl(url: string): ValueBackendEndpointSpec<void>;
+    setActiveActions(actions: string[]): ValueBackendEndpointSpec<void>;
+    setTitle(title: string): ValueBackendEndpointSpec<void>;
+    pickFile(type?: string): ValueBackendEndpointSpec<ExternalFile | null>;
 }>;
 
-type InfoEndpoint = ObjectEndpointSpec<{
-    getRuntime(): ValueEndpointSpec<string>;
-    getVersion(): ValueEndpointSpec<string>;
+type InfoBackend = ObjectBackendEndpointSpec<{
+    getRuntime(): ValueBackendEndpointSpec<string>;
+    getVersion(): ValueBackendEndpointSpec<string>;
 }>;
 
-type EndpointEndpoint = ObjectEndpointSpec<{
+type EndpointBackend = ObjectBackendEndpointSpec<{
```

Comment:
I'm a bit confused how this is `ObjectBackendEndpointSpec` now (from previously `ObjectEndpointSpec`). Wasn't the idea to get rid of the term _endpoint_ within postoffice - except for the new _endpoint_ API - and use the term _backend_ everywhere instead? That'd give us `ObjectBackendSpec` etc, not `ObjectEndpointBackendSpec`. Otherwise, the line I've left this comment on still seems pretty confusing to me :D

Reply:
Yeah, that's a fair point, but it felt a bit odd creating a backend object and backendEndpoint encapsulated the functionality well. But if its still confusing then I can change this to Backend alone